xbmc/network/DNSNameCache.cpp

```diff
@@ -84,20 +109,42 @@ bool CDNSNameCache::Lookup(const std::string& strHostName, std::string& strIpAdd
 #endif
 
   // perform dns lookup
-  struct hostent *host = gethostbyname(strHostName.c_str());
-  if (host && host->h_addr_list[0])
+  struct addrinfo hints;
+  struct addrinfo *result(NULL);
+
+  memset(&hints, 0, sizeof(struct addrinfo));
+  int err;
+
+  // prefer DNS record type (A vs AAAA)  be this of active interface(address)
+  // otherwise (by default) system prefers A(IPv4) records and this will make
+  // troubles on dual configured hosts - but without wide area access over IPv4
+  do {
+    freeaddrinfo(result);
+    hints.ai_family = hints.ai_family == 0 ? g_application.getNetwork().GetFirstConnectedFamily() : AF_UNSPEC;
+    err = getaddrinfo(strHostName.c_str(), NULL, &hints, &result);
+  } while ((err || !result) && hints.ai_family != AF_UNSPEC);
+
+  std::string str_err;
+  if (err)
+    str_err = gai_strerror(err);
+
+  bool bReturn(result != NULL);
```

Comment:
you remember Pascal? and Fortran or DBase? maybe
everything I know about c++ is from xbmc and libcec code. based on your protest I would say more from libcec :)